### PR TITLE
Add provision killswitch

### PR DESCRIPTION
**Pull request by m00g3n**
**Description**

Changes proposed in this pull request:

- Add provision kill switch to prevent creating clusters that are handled by the provisioner
- `runtime-reconciler-enabled` flag default value is `true` now 

**Related issue(s)**

- https://github.com/kyma-project/infrastructure-manager/issues/250

### CODE REVIEW

File: internal/controller/runtime/fsm/runtime_fsm_create_shoot_dry_run.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		return updateStatePendingWithErrorAndStop(
+			&s.instance,
+			imv1.ConditionTypeRuntimeProvisioned,
+			imv1.ConditionReasonConversionError,
+			"Runtime conversion error")
+	}
+
+	s.shoot = &newShoot
+	s.instance.UpdateStateReady(
+		imv1.ConditionTypeRuntimeProvisioned,
```
Comment (Disper) [inline internal/controller/runtime/fsm/runtime_fsm_create_shoot_dry_run.go:26]: if someone filters through type=Provisioned, that person might thing that it was actually done by KIM.

How about having a separate type, such as `ProvisionedDryRun`?

File: internal/controller/runtime/runtime_controller.go [hunk 2]
```diff
@@ -91,6 +91,9 @@ func (r *RuntimeReconciler) UpdateShootClient(client client.Client) {
 func (r *RuntimeReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&imv1.Runtime{}).
-		WithEventFilter(predicate.GenerationChangedPredicate{}).
+		WithEventFilter(predicate.Or(
+			predicate.GenerationChangedPredicate{},
```
[the remark this answers is not in the document]
Comment (m00g3n) [reply internal/controller/runtime/runtime_controller.go:95]: We want to be able to control which application will do the provisioning on runtime (e.g. when someone will add/change/remove `kyma-project.io/controlled-by-provisioner` label)

File: docs/README.md [hunk 3]
```diff
@@ -16,14 +16,18 @@ You can configure the Infrastructure Manager deployment with the following argum
...
-5. `runtime-reconciler-enabled` - feature flag responsible for enabling the runtime reconciler. Default value is `false`.
+5. `runtime-reconciler-enabled` - feature flag responsible for enabling the runtime reconciler. Default value is `true`.
 6. `shoot-spec-dump-enabled` - feature flag responsible for enabling the shoot spec dump. Default value is `false`.
 
 
 See [manager_gardener_secret_patch.yaml](../config/default/manager_gardener_secret_patch.yaml) for default values.
 
 ## Troubleshooting
 
+1. Switching between the `provisioner` and the `kim`.
```
Comment (grego952) [inline docs/README.md:27]:
```suggestion
1. Switching between the `provisioner` and `kim`.
```

File: docs/README.md [hunk 4]
```diff
@@ -16,14 +16,18 @@ You can configure the Infrastructure Manager deployment with the following argum
...
+
+The `kyma-project.io/controlled-by-provisioner` label provides fine-grained control over `Runtime` CR. Only if the label value is set to `false`, resource is considered managed and will be controlled by the `kyma-application-manager`.
```
Comment (grego952) [inline docs/README.md:29]:
```suggestion
The `kyma-project.io/controlled-by-provisioner` label provides fine-grained control over the `Runtime` CR. Only if the label value is set to `false`, the resource is considered managed and will be controlled by `kyma-application-manager`.
```

**Review by grego952 — APPROVED** [review]

**Pull request merged**